# Profitable App Profiles for the App Store and Google Play Markets

This project is to find mobile app profiles that are profitable for the <h3 id="https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home">App Store</h3>  and <h3 id="https://www.kaggle.com/lava18/google-play-store-apps/home">Google Play</h3> markets. As data analysts for a company that builds Android and iOS mobile apps, this analysis is to enable developers to make data-driven decisions with respect to the kind of apps they build.

Apps that are free to download and install, revenue for any given app is mostly influenced by the number of users that use app. This project is to analyze data to help developers understand what kinds of apps are likely to attract more users.

In [1]:
from csv import reader
#open and read apple store data
open_file = open('AppleStore.csv')
read_file= reader(open_file)
ios_data = list(read_file)
ios_header = ios_data[0]
ios = ios_data[1:]
#open and read google play data
open_file = open('googleplaystore.csv')
read_file= reader(open_file)
android_data = list(read_file)
android_header = android_data[0]
android = android_data[1:]

print(ios_header,'\n')
print(android_header, '\n')

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'] 

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 



In [2]:
len(android)

10841

The function explore_data() that can repeatedly use to print rows in a readable way.

In [3]:
def explore_data(dataset, start, end, row_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row, '\n')
        
    if row_and_columns:
        print('No of rows: ', len(dataset))
        print('No of columns: ', len(dataset[0]))

print(android_header,'\n')
explore_data(android,0,3,True)
print('\n')
print(ios_header,'\n')
explore_data(ios,0,3,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'] 

['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up'] 

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'] 

No of rows:  10841
No of columns:  13


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'] 

['284882215', '

Checking every row in the android data whether it has the same no of data points as the header. By which we could find out the missing data points of that row and remove that row from the dataset.

In [4]:
for row in android:
    if len(row)!=len(android_header):
        print(row)
        print(android_data.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10473


In [5]:
print(len(android[10473]))
print(android[10473],'\n')
print(android_header,'\n')
print(android[0],'\n')

13
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up'] 

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'] 



The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5. As a consequence,we'll delete this row.

In [6]:
print(len(android))
del android[10472]  # don't run this more than once
print(len(android))

10841
10840


notice some apps have duplicate entries. For instance, Instagram has four entries:

In [7]:
for app_name in android:
    name = app_name[0]
    if name == 'Instagram':
        print(app_name, '\n')

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device'] 

['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device'] 

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device'] 

['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device'] 



We need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, or in this casethe main difference happens on the fourth position(index 3) of each row, which corresponds to the number of reviews.



In [8]:
#creating lists for the duplicate and unique apps

duplicate_apps = []
unique_apps = []

for app_name in android:
    name = app_name[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        

print('length of duplicate app list',len(duplicate_apps),'\n')
print(duplicate_apps[0:5],'\n')
print('No of unique apps:', len(unique_apps),'\n')

length of duplicate app list 1181 

['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings'] 

No of unique apps: 9659 



In total, there are 1,181 cases where an app occurs more than once:

We can use this information to build a criterion for removing the duplicates. The higher the number of reviews (for instagram apps), the more recent the data should be. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries for any given app.

Checking IOS duplicate apps

In [9]:
ios_duplicate_apps = []
ios_unique_apps = []

for apps in ios:
    name = apps[1]
    if name in ios_unique_apps:
        ios_duplicate_apps.append(name)
    else:
        ios_unique_apps.append(name)
        
print('length of duplicate app list',len(ios_duplicate_apps),'\n')
print(ios_duplicate_apps,'\n')
print('No of unique apps:', len(ios_unique_apps),'\n')

for apps in ios:
    name = apps[1]
    if name in ios_duplicate_apps:
        print(apps,'\n')

length of duplicate app list 2 

['Mannequin Challenge', 'VR Roller Coaster'] 

No of unique apps: 7195 

['1173990889', 'Mannequin Challenge', '109705216', 'USD', '0.0', '668', '87', '3.0', '3.0', '1.4', '9+', 'Games', '37', '4', '1', '1'] 

['952877179', 'VR Roller Coaster', '169523200', 'USD', '0.0', '107', '102', '3.5', '3.5', '2.0.0', '4+', 'Games', '37', '5', '1', '1'] 

['1178454060', 'Mannequin Challenge', '59572224', 'USD', '0.0', '105', '58', '4.0', '4.5', '1.0.1', '4+', 'Games', '38', '5', '1', '1'] 

['1089824278', 'VR Roller Coaster', '240964608', 'USD', '0.0', '67', '44', '3.5', '4.0', '0.81', '4+', 'Games', '38', '0', '1', '1'] 



To remove the duplicates, create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app.

In [10]:
reviews_max = {}

for apps in android:
    name = apps[0]
    n_reviews = float(apps[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [11]:
print('Length of dictionary which have max no of data point: ',len(reviews_max))
print('No of unique apps:', len(unique_apps),'\n')

Length of dictionary which have max no of data point:  9659
No of unique apps: 9659 



length of dic and no of unique apps value are same so now we can use this dictionary to create dataset without having any duplicate value on it.

In [12]:
ios_size_max = {}

for apps in ios:
    name = apps[1]
    size = apps[2]
    if name in ios_size_max and ios_size_max[name] < size:
        ios_size_max[name] = size
    elif name not in ios_size_max:
        ios_size_max[name] = size
        
print('Length of dictionary which have max no of data point: ',len(ios_size_max))
print('No of unique apps:', len(ios_unique_apps),'\n')

Length of dictionary which have max no of data point:  7195
No of unique apps: 7195 



In [13]:
#check the dic
reviews_max

{'Paramedic Protocol Provider': 171.0,
 'PlayKids - Educational cartoons and games for kids': 182173.0,
 'AppClose - The #1 FREE Co-Parenting App': 159.0,
 'ck-modelcars Shop': 118.0,
 'CJ: Strike Back': 29330.0,
 'The Divine Feminine App: the DF App': 8.0,
 'Билеты ПДД CD 2019 PRO': 21.0,
 'Content Transfer': 19302.0,
 'ZipRealty Real Estate & Homes': 3937.0,
 "Five Nights at Freddy's 2 Demo": 1503544.0,
 'cm, mm to inch, feet converter tool': 176.0,
 'Rejoin Your Ex': 0.0,
 'Elevate - Brain Training Games': 248912.0,
 'Hotspot Shield Free VPN Proxy & Wi-Fi Security': 1116393.0,
 'Resume ( CV Editor )': 416.0,
 'Roll the Ball® - slide puzzle': 1385093.0,
 'Daily Murli Saar Widget': 308.0,
 'Rabo Banking': 31906.0,
 'DSLR camera - Auto Focus and Blur Professional': 30.0,
 'SCRABBLE': 172281.0,
 'Chase Mobile': 1374549.0,
 'FC BARCELONA EVENTS': 6.0,
 'magicApp Calling & Messaging': 207712.0,
 'Wendy’s – Food and Offers': 6512.0,
 'Bitcoin BX Thailand': 120.0,
 'Eh Amego!': 13388.0,
 'E

In [14]:
ios_size_max

{'Gate Keepers': '107710464',
 'Yoshirt - Design Your Own Custom Tshirt, Tote Bag, Socks and More': '49271808',
 '僕の魔球打てるの？': '20328448',
 'Nonstop Knight': '161026048',
 'CARS MOD FOR MINECRAFT PC EDITION - POCKET MINE GUIDE': '24840192',
 'Sync Solver for Fitbit': '14583808',
 "Reiner Knizia's Tigris & Euphrates": '54852324',
 'Mad City Gangs: Nice City': '164147200',
 'Sky Pillar': '86454272',
 'The Blacklist: Conspiracy': '101144576',
 'ゲオ\u3000クーポンが貰える！ゲーム予約もできる！': '34163712',
 'Winter Fugitives 2: Chronicles': '166257664',
 '【謎解き】自称カノジョからの病みすぎメッセージ': '16397312',
 'Fear for Sale: The House on Black River (Full)': '2233358336',
 'Ketchapp Tennis': '167264256',
 'Kinectimals': '185228498',
 "Donald's Empire": '295349248',
 '金のフレーズ 〜「新TOEIC®TEST 出る単特急 金のフレーズ」収録 〜': '85052416',
 'Glob Trotters - Endless Arcade Blobber': '136354816',
 'Odysseus: Long Way Home HD (Full)': '281676680',
 'Chase Mobile': '34950144',
 'ケンタッキーフライドチキン\u3000公式アプリ': '50291712',
 '小学館 オックスフォード 英語コロケーション辞典': '720

Use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews). Use dictionary to remove the duplicate rows.

We start by initializing two empty lists, android_clean and already_added.
###We loop through the android data set, and for every iteration:
###We isolate the name of the app and the number of reviews.
We add the current row (app) to the android_clean list, and the app name (name) to the already_cleaned list if:
####The number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary.
####The name of the app is not already in the already_added list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for reviews_max[name] == n_reviews, we'll still end up with duplicate entries for some apps.

In [15]:
android_clean = [] #which will store our new cleaned data set
already_added = [] #which will just store app names

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
print(len(android_clean))

9659


In [16]:
ios_clean = []
ios_already_add = []

for app in ios:
    name = app[1]
    size = app[2]
    if (size == ios_size_max[name]) and (name not in ios_already_add):
        ios_clean.append(app)
        ios_already_add.append(name)
        
print(len(ios_clean))

7195


In [17]:
for row in android_clean:
    name = row[0]
    if name == 'Box':  #check with Instagram as well
        print(row)
        

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


From instagram/Box rows above example we could be ensured that in android_clean list we 
have only unique rows with max no of reviews

In [18]:
def non_english_app (app_name_check):
    for each_character in app_name_check:
        if ord(each_character)>127:
            return False
    
    return True

In [19]:
non_english_app('Instagram')

True

In [20]:
non_english_app('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [21]:
non_english_app('Docs To Go™ Free Office Suite')

False

In [22]:
non_english_app('Instachat 😜')

False

The function seems to work fine, but some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, we'll remove useful apps if we use the function in its current form.

To address the issue right now rewrite the function in a way that input string has more than three characters that fall outside the ASCII range (0 - 127), then the function should return False (identify the string as non-English), otherwise it should return True.

In [23]:
def is_english_app(input_string):
    ascii_count = 0
    for each in input_string:
        if ord(each)>127:
            ascii_count +=1
    if ascii_count>3:
        return False
    return True

In [24]:
is_english_app('Docs To Go™ Free Office Suite')

True

In [25]:
is_english_app('Instachat 😜')

True

In [26]:
android_english = []
ios_english = []

for row in android_clean:
    name = row[0]
    if is_english_app(name):
        android_english.append(row)
        
print('Without Duplicate English apps Google PlayStore data: ',len(android_english),'\n')

for row in ios_clean:
    name = row[0]
    if is_english_app(name):
        ios_english.append(row)
        
print('Without Duplicate English apps Apple Store data: ',len(ios_english),'\n')

Without Duplicate English apps Google PlayStore data:  9614 

Without Duplicate English apps Apple Store data:  7195 



In [27]:
explore_data(android_english,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'] 

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'] 

['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up'] 

No of rows:  9614
No of columns:  13


In [28]:
explore_data(ios_english,0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'] 

['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'] 

['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'] 

No of rows:  7195
No of columns:  16


In [29]:
android_free_apps = []
ios_free_apps = []
#find out and make a seperate lists for the apps which are quite free
for apps in android_english:
    price = apps[7]
    if price == '0':
        android_free_apps.append(apps)
        
for apps in ios_english:
    price = apps[4]
    if price == '0.0':
        ios_free_apps.append(apps)
        
print("No of Android Free aps: ", len(android_free_apps), " "
     "No of IOS Free apss: ", len(ios_free_apps))        

No of Android Free aps:  8864  No of IOS Free apss:  4054


Project aim is to determine the kinds of apps that are likely to attract more users because free apps revenue is highly influenced by the number of people using that apps.

To do analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the  prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

In [30]:
print('IOS Prime genres index no: ',ios_header.index('prime_genre'))
print('Android Genres index no: ',android_header.index('Genres'))
print('Android Category index no: ',android_header.index('Category'))

IOS Prime genres index no:  11
Android Genres index no:  9
Android Category index no:  1


The following function to generate frequency tables that show percentages

In [31]:
def freq_table(dataset, index):
    freq_table = {}
    freq_table_percentage = {}
    total = 0
    for row in dataset:
        name = row[index]
        if name in freq_table:
            freq_table[name] +=1
        elif name not in freq_table:
            freq_table[name] =1
        
        total +=1
    
    for value in freq_table:
        percentage = round(((freq_table[value]/total)*100),2)
        freq_table_percentage[value] = percentage
        
    return freq_table_percentage

In [32]:
freq_table(ios_free_apps,11)

{'Book': 1.63,
 'Business': 0.49,
 'Catalogs': 0.22,
 'Education': 3.26,
 'Entertainment': 8.24,
 'Finance': 2.07,
 'Food & Drink': 1.06,
 'Games': 55.62,
 'Health & Fitness': 1.87,
 'Lifestyle': 2.32,
 'Medical': 0.2,
 'Music': 1.65,
 'Navigation': 0.49,
 'News': 1.43,
 'Photo & Video': 4.12,
 'Productivity': 1.53,
 'Reference': 0.49,
 'Shopping': 2.98,
 'Social Networking': 3.53,
 'Sports': 1.95,
 'Travel': 1.38,
 'Utilities': 2.69,
 'Weather': 0.76}

In [33]:
freq_table(android_free_apps,9)

{'Action': 3.1,
 'Action;Action & Adventure': 0.1,
 'Adventure': 0.68,
 'Adventure;Action & Adventure': 0.03,
 'Adventure;Education': 0.01,
 'Arcade': 1.85,
 'Arcade;Action & Adventure': 0.12,
 'Arcade;Pretend Play': 0.01,
 'Art & Design': 0.6,
 'Art & Design;Action & Adventure': 0.01,
 'Art & Design;Creativity': 0.07,
 'Art & Design;Pretend Play': 0.01,
 'Auto & Vehicles': 0.93,
 'Beauty': 0.6,
 'Board': 0.38,
 'Board;Action & Adventure': 0.02,
 'Board;Brain Games': 0.08,
 'Books & Reference': 2.14,
 'Books & Reference;Education': 0.01,
 'Business': 4.59,
 'Card': 0.45,
 'Card;Action & Adventure': 0.01,
 'Casino': 0.43,
 'Casual': 1.76,
 'Casual;Action & Adventure': 0.14,
 'Casual;Brain Games': 0.14,
 'Casual;Creativity': 0.07,
 'Casual;Education': 0.02,
 'Casual;Music & Video': 0.01,
 'Casual;Pretend Play': 0.24,
 'Comics': 0.61,
 'Comics;Creativity': 0.01,
 'Communication': 3.24,
 'Communication;Creativity': 0.01,
 'Dating': 1.86,
 'Education': 5.35,
 'Education;Action & Adventure':

In [34]:
freq_table(android_free_apps,1)

{'ART_AND_DESIGN': 0.64,
 'AUTO_AND_VEHICLES': 0.93,
 'BEAUTY': 0.6,
 'BOOKS_AND_REFERENCE': 2.14,
 'BUSINESS': 4.59,
 'COMICS': 0.62,
 'COMMUNICATION': 3.24,
 'DATING': 1.86,
 'EDUCATION': 1.16,
 'ENTERTAINMENT': 0.96,
 'EVENTS': 0.71,
 'FAMILY': 18.91,
 'FINANCE': 3.7,
 'FOOD_AND_DRINK': 1.24,
 'GAME': 9.72,
 'HEALTH_AND_FITNESS': 3.08,
 'HOUSE_AND_HOME': 0.82,
 'LIBRARIES_AND_DEMO': 0.94,
 'LIFESTYLE': 3.9,
 'MAPS_AND_NAVIGATION': 1.4,
 'MEDICAL': 3.53,
 'NEWS_AND_MAGAZINES': 2.8,
 'PARENTING': 0.65,
 'PERSONALIZATION': 3.32,
 'PHOTOGRAPHY': 2.94,
 'PRODUCTIVITY': 3.89,
 'SHOPPING': 2.25,
 'SOCIAL': 2.66,
 'SPORTS': 3.4,
 'TOOLS': 8.46,
 'TRAVEL_AND_LOCAL': 2.34,
 'VIDEO_PLAYERS': 1.79,
 'WEATHER': 0.8}

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.



Another function we can use to display the percentages in a descending order

To do that, we'll need to make use of the built-in sorted() function. This function takes in an iterable data type (like a list, dictionary, tuple, etc.), and returns a list of the elements of that iterable sorted in ascending or descending order (the reverse parameter controls whether the order is ascending or descending).

The sorted() function doesn't work too well with dictionaries because it only considers and returns the dictionary keys.

However, the sorted() function works well if we transform the dictionary into a list of tuples, where each tuple contains a dictionary key along with its corresponding dictionary value. To ensure the sorting works right, the dictionary value comes first, and the dictionary key comes second.

In [35]:
'''function we can use to display the percentages dictionary
in a descending order'''

def display_table (dataset, index):
    table = freq_table(dataset,index)
    table_display = []
    for key in table:
        key_value_as_tuple = (table[key], key)
        table_display.append(key_value_as_tuple)
    
    sorted_table = sorted(table_display, reverse=True)
    
    for each_category in sorted_table:
        print(each_category[1], ': ', each_category[0])

displaying and checking the ios and android free apss in decending order
using display table function. 

In [36]:
display_table(ios_free_apps,11)

Games :  55.62
Entertainment :  8.24
Photo & Video :  4.12
Social Networking :  3.53
Education :  3.26
Shopping :  2.98
Utilities :  2.69
Lifestyle :  2.32
Finance :  2.07
Sports :  1.95
Health & Fitness :  1.87
Music :  1.65
Book :  1.63
Productivity :  1.53
News :  1.43
Travel :  1.38
Food & Drink :  1.06
Weather :  0.76
Reference :  0.49
Navigation :  0.49
Business :  0.49
Catalogs :  0.22
Medical :  0.2


In [37]:
display_table(android_free_apps,1)

FAMILY :  18.91
GAME :  9.72
TOOLS :  8.46
BUSINESS :  4.59
LIFESTYLE :  3.9
PRODUCTIVITY :  3.89
FINANCE :  3.7
MEDICAL :  3.53
SPORTS :  3.4
PERSONALIZATION :  3.32
COMMUNICATION :  3.24
HEALTH_AND_FITNESS :  3.08
PHOTOGRAPHY :  2.94
NEWS_AND_MAGAZINES :  2.8
SOCIAL :  2.66
TRAVEL_AND_LOCAL :  2.34
SHOPPING :  2.25
BOOKS_AND_REFERENCE :  2.14
DATING :  1.86
VIDEO_PLAYERS :  1.79
MAPS_AND_NAVIGATION :  1.4
FOOD_AND_DRINK :  1.24
EDUCATION :  1.16
ENTERTAINMENT :  0.96
LIBRARIES_AND_DEMO :  0.94
AUTO_AND_VEHICLES :  0.93
HOUSE_AND_HOME :  0.82
WEATHER :  0.8
EVENTS :  0.71
PARENTING :  0.65
ART_AND_DESIGN :  0.64
COMICS :  0.62
BEAUTY :  0.6


According to ios data frequency table Games is the most common genre, Entertainment, photo videos, social networking are follwed by it. 

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:



In [38]:
ios_genres = freq_table(ios_free_apps, 11)

for genre in ios_genres:
    total = 0
    len_genre = 0
    for apps in ios_free_apps:
        genre_app = apps[11]
        if genre_app == genre:
            tot_n_rating = float(apps[5])
            total += tot_n_rating
            len_genre +=1
    #print(len_genre)
    #if we print the len_genre then for every genre no of apps are shown
    # manually if we sum all the no of apps then it will 4054 as same
    #No of IOS Free apss:  4054
    avg_n_usr_rating = round((total / len_genre),2)
    print(genre, ': ', avg_n_usr_rating)

Food & Drink :  20179.09
Weather :  47220.94
Education :  6266.33
Productivity :  19053.89
Medical :  459.75
Travel :  20216.02
Business :  6367.8
Photo & Video :  27249.89
Lifestyle :  8978.31
Reference :  67447.9
Games :  18941.13
News :  15892.72
Sports :  20128.97
Music :  56482.03
Finance :  13522.26
Utilities :  14010.1
Navigation :  25972.05
Shopping :  18746.68
Book :  8498.33
Health & Fitness :  19952.32
Social Networking :  53078.2
Catalogs :  1779.56
Entertainment :  10822.96


Reference has the max no of user reviews, onwards music, social network

In [39]:
for app in ios_free_apps:
    if app[11] == 'Reference':
        print(app[1], ':', app[5]) #index 5 rating_count_tot 

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
彩库宝典-【官方版】 : 0
Jishokun-Japanese English Dictionary & Translator : 0
無料で音楽や写真・カメラの裏技アプリ for iPhone7 : 0


In [40]:
for app in ios_free_apps:
    if app[11] == 'Social Networking':
        print(app[1], ':', app[5]) #index 5 rating_count_tot

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [41]:
for app in ios_free_apps:
    if app[11] == 'Music':
        print(app[1], ':', app[5]) #index 5 rating_count_tot

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

The same pattern applies to music and social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.




However, this niche on Reference seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.)

In [42]:
display_table(android_free_apps,5)

1,000,000+ :  15.73
100,000+ :  11.55
10,000,000+ :  10.55
10,000+ :  10.2
1,000+ :  8.39
100+ :  6.92
5,000,000+ :  6.83
500,000+ :  5.56
50,000+ :  4.77
5,000+ :  4.51
10+ :  3.54
500+ :  3.25
50,000,000+ :  2.3
100,000,000+ :  2.13
50+ :  1.92
5+ :  0.79
1+ :  0.51
500,000,000+ :  0.27
1,000,000,000+ :  0.23
0+ :  0.05
0 :  0.01


For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to convert each install number from string to float by replacing comma and plus character using built in 
str.replace() function.

In [43]:
android_genres = freq_table(android_free_apps, 1)

for category in android_genres:
    total = 0
    len_category = 0
    for app in android_free_apps:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
            
    avg_n_installs = round((total / len_category),2)
    print(category, ':', avg_n_installs)

BOOKS_AND_REFERENCE : 8767811.89
COMMUNICATION : 38456119.17
LIBRARIES_AND_DEMO : 638503.73
ART_AND_DESIGN : 1986335.09
BUSINESS : 1712290.15
LIFESTYLE : 1437816.27
TRAVEL_AND_LOCAL : 13984077.71
PERSONALIZATION : 5201482.61
COMICS : 817657.27
WEATHER : 5074486.2
SHOPPING : 7036877.31
PHOTOGRAPHY : 17840110.4
PRODUCTIVITY : 16787331.34
SOCIAL : 23253652.13
HEALTH_AND_FITNESS : 4188821.99
MEDICAL : 120550.62
GAME : 15588015.6
FAMILY : 3695641.82
TOOLS : 10801391.3
VIDEO_PLAYERS : 24727872.45
NEWS_AND_MAGAZINES : 9549178.47
BEAUTY : 513151.89
ENTERTAINMENT : 11640705.88
HOUSE_AND_HOME : 1331540.56
DATING : 854028.83
EVENTS : 253542.22
AUTO_AND_VEHICLES : 647317.82
SPORTS : 3638640.14
FOOD_AND_DRINK : 1924897.74
PARENTING : 542603.62
EDUCATION : 1833495.15
MAPS_AND_NAVIGATION : 4056941.77
FINANCE : 1387692.48


On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs.

In [44]:
for app in android_free_apps:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times.

In [45]:
under_100_m = []

for app in android_free_apps:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
round((sum(under_100_m) / len(under_100_m)),2)

3603485.39

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

In [46]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:



In [47]:
for app in android_free_apps:
    if app[1]== 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
                    print(app[0],': ', app[5])

Wikipedia :  10,000,000+
Cool Reader :  10,000,000+
Book store :  1,000,000+
FBReader: Favorite Book Reader :  10,000,000+
Free Books - Spirit Fanfiction and Stories :  1,000,000+
AlReader -any text book reader :  5,000,000+
FamilySearch Tree :  1,000,000+
Cloud of Books :  1,000,000+
ReadEra – free ebook reader :  1,000,000+
Ebook Reader :  5,000,000+
Read books online :  5,000,000+
eBoox: book reader fb2 epub zip :  1,000,000+
All Maths Formulas :  1,000,000+
Ancestry :  5,000,000+
HTC Help :  10,000,000+
Moon+ Reader :  10,000,000+
English-Myanmar Dictionary :  1,000,000+
Golden Dictionary (EN-AR) :  1,000,000+
All Language Translator Free :  1,000,000+
Aldiko Book Reader :  10,000,000+
Dictionary - WordWeb :  5,000,000+
50000 Free eBooks & Free AudioBooks :  5,000,000+
Al-Quran (Free) :  10,000,000+
Al Quran Indonesia :  10,000,000+
Al'Quran Bahasa Indonesia :  10,000,000+
Al Quran Al karim :  1,000,000+
Al Quran : EAlim - Translations & MP3 Offline :  5,000,000+
Koran Read &MP3 30

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.